### PR TITLE
Switch buildbot to using CPack for packaging

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

**Comment by steven-johnson** [comment]
Seems to be working, PTAL

File: master/master.cfg [hunk 1]
```diff
@@ -672,6 +682,70 @@ def add_halide_cmake_build_steps(factory, builder_type, configs):
...
+def add_halide_cmake_package_steps(factory, builder_type, configs):
+    for config in configs:
+        assert config == 'Release'
+
+        source_dir = get_halide_source_path()
+
+        # TODO: this is clumsy; we assume an integer version for llvm
+        # and also assume the Halide version matches; this is sorta
+        # true now (for Halide trunk) but needs more sophistication
+        # once we make Halide release branches
```
Comment (alexreinking) [inline master/master.cfg:694]: Can we open an issue on this repo about this?

File: master/master.cfg [hunk 2]
```diff
@@ -672,6 +682,70 @@ def add_halide_cmake_build_steps(factory, builder_type, configs):
...
+        env = get_env(builder_type)
+        env['Clang_DIR'] = get_llvm_install_path(config, 'lib/cmake/clang')
+        env['LLVM_DIR'] = get_llvm_install_path(config, 'lib/cmake/llvm')
+        env['Halide_VERSION'] = version
+
+        if builder_type.os == 'windows':
+            # TODO: on Windows, we can't use Ninja for packaging (as we do everywhere
+            # else in this cfg) due to a bug in CMake 3.18, so we must use MSBuild;
+            # that means we must use a different build directory entirely. To simplify the
+            # world, we make this a subdir of the real build dir (so it gets cleaned properly).
```
Comment (alexreinking) [inline master/master.cfg:709]: Same deal here... open an issue. I'll file a bug with the CMake folks; assign it to me.

File: master/master.cfg [hunk 3]
```diff
@@ -393,16 +399,18 @@ def get_msvc_config_steps(factory, builder_type):
...
     rev = props.getProperty('got_revision')['halide']
-    builder = props.getProperty('buildername')
-    if 'windows' in str(builder):
-        suffix = '.zip'
-    else:
-        suffix = '.tgz'
+    builder_label = props.getProperty('buildername')
+
     # Always upload to /home/abadams/artifacts -- regardless of user --
     # since that's there the public-facing webpage looks
```
Comment (alexreinking) [inline master/master.cfg:407]: `s/there/where`?

**Review by alexreinking — APPROVED** [review]

**Pull request merged**